File: Codes/Hash/Non-Cryptographic/RSHash/RSHash.vb
'  Hash function by Robert Sedgwicks
'  Archive of Reversing.ID
'  Non-Cryptographic Hash
'

' Fungsi ini merupakan method dari sebuah class
Public Shared Function RSHash (key As String) As UInteger
    Dim b As UInteger = 378551
	Dim a As ULong = 63689
	Dim state As ULong = 0
	Dim i As UInteger = 0

	For i = 0 To key.Length - 1
		state = (state * a + CByte(AscW(key(CInt(i))))) And UInteger.MaxValue
		a = (a * b) And UInteger.MaxValue
	Next

	Return state
End Function

File: Codes/Hash/Non-Cryptographic/RSHash/RSHash.cs
/*
	Hash function by Robert Sedgwicks
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static uint RSHash (string key)
{
    uint b = 378551;
	uint a = 63689;
	uint state = 0;
	uint i = 0;

	for (i = 0; i < key.Length; i++)
	{
		state = state * a + ((byte)key[(int)i]);
		a *= b;
	}

	return state;
}